modules/rules_foreign_cc/0.7.1/patches/label.patch

```diff
@@ -0,0 +1,159 @@
+-    if not absolute_label.startswith("@"):
+-        absolute_label = "@rules_foreign_cc" + absolute_label
+-
+     repository_ctx.file("defs.bzl", _DEFS_BZL_CONTENT.format(
+-        commands_src = absolute_label,
++        commands_src = repository_ctx.attr.commands_src,
+         symbols = "\n    ".join(["\"{}\",".format(symbol) for symbol in PLATFORM_COMMANDS.keys()]),
+         commands = "\n    ".join(["{cmd} = {cmd},".format(cmd = symbol) for symbol in PLATFORM_COMMANDS.keys()]),
+     ))
+@@ -72,9 +66,8 @@ framework_toolchain_repository = repository_rule(
+     doc = "A repository rule which defines a `@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain` toolchain.",
+     implementation = _framework_toolchain_repository_impl,
+     attrs = {
+-        "commands_src": attr.label(
+-            doc = "The label of a `.bzl` source which defines toolchain commands",
+-            allow_files = [".bzl"],
++        "commands_src": attr.string(
++            doc = "The string of a `.bzl` source which defines toolchain commands",
+         ),
+         "exec_compatible_with": attr.string_list(
+             doc = "A list of constraint_values that must be present in the execution platform for this target.",
+@@ -96,7 +89,7 @@ def register_framework_toolchains(register_toolchains = True):
+ 
+     for item in TOOLCHAIN_MAPPINGS:
+         # Generate a toolchain name without the `.bzl` suffix
+-        toolchain_name = "rules_foreign_cc_framework_toolchain_" + item.file.name[:-len(".bzl")]
++        toolchain_name = "rules_foreign_cc_framework_toolchain_" + item.file.split(":")[1][:-len(".bzl")]
+ 
+         framework_toolchain_repository(
+             name = toolchain_name,
+diff --git a/foreign_cc/private/framework/toolchains/access.bzl b/foreign_cc/private/framework/toolchains/access.bzl
+index 16d05d1..3fb5988 100644
+--- a/foreign_cc/private/framework/toolchains/access.bzl
++++ b/foreign_cc/private/framework/toolchains/access.bzl
+@@ -16,7 +16,7 @@ def create_context(ctx):
+             - prelude (dict): A cache for rendered functions
+     """
+     return struct(
+-        shell = ctx.toolchains[str(Label("//foreign_cc/private/framework:shell_toolchain"))].commands,
++        shell = ctx.toolchains["@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain"].commands,
+         prelude = {},
+     )
+ 
+diff --git a/foreign_cc/private/framework/toolchains/mappings.bzl b/foreign_cc/private/framework/toolchains/mappings.bzl
+index ff7849b..4a9a75d 100644
+--- a/foreign_cc/private/framework/toolchains/mappings.bzl
++++ b/foreign_cc/private/framework/toolchains/mappings.bzl
+@@ -23,24 +23,24 @@ TOOLCHAIN_MAPPINGS = [
+         exec_compatible_with = [
+             "@platforms//os:linux",
+         ],
+-        file = Label("@rules_foreign_cc//foreign_cc/private/framework/toolchains:linux_commands.bzl"),
++        file = "@rules_foreign_cc//foreign_cc/private/framework/toolchains:linux_commands.bzl",
```

Comment:
these are the dubious ones, since `command_src` is now a string attribute. If some other module has sees rules_foreign_cc under a different repo name and uses these mappings, they'd likely get reference errors.

EDIT: see EDIT in the other comment above -- this might actually work fine.